### PR TITLE
Return ObjectCount and validations only of current ns

### PR DESCRIPTION
**Pull request by xeviknal**
Backend part of this one: https://github.com/kiali/kiali-ui/pull/1646

### CODE REVIEW

**Comment by xeviknal** [comment]
@jshaughn it passes the swagger check now. Thank you for checking this out :)

File: models/istio_validation_test.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func TestSummarizeValidations(t *testing.T) {
 		},
 	}
 
-	summary := validations.SummarizeValidation()
+	summary := validations.SummarizeValidation("bookinfo")
 
 	assert.Equal(2, summary.Warnings)
 	assert.Equal(2, summary.Errors)
```
Comment (jotak) [inline models/istio_validation_test.go]: would be nice to add an assertion on ObjectCount too

File: models/istio_validation.go [hunk 2]
```diff
@@ -314,10 +318,12 @@ func (iv IstioValidations) MergeReferences(validations IstioValidations) IstioVa
 	return iv
 }
 
-func (iv IstioValidations) SummarizeValidation() IstioValidationSummary {
+func (iv IstioValidations) SummarizeValidation(ns string) IstioValidationSummary {
 	ivs := IstioValidationSummary{}
```
Comment (jotak) [inline models/istio_validation.go]: maybe here there should be an explicit initialization? (see my comment on frontend PR)

Comment (xeviknal) [reply models/istio_validation.go]: Default are 0 in this case. Then, the empty object looks like this:
```json
{
"errors": 0,
"objectCount": 0,
"warnings": 0
}
```
Does it make sense not to add any initial data?

Comment (jotak) [reply models/istio_validation.go]: yep, that's fine ; tbh I'm sometimes puzzled with go handling of zero-value objects and i wasn't sure if that would come as 0 or undefined on serialization. It's fine to keep as you did.

File: models/istio_validation.go [hunk 3]
```diff
@@ -329,6 +335,7 @@ func (summary *IstioValidationSummary) mergeSummaries(cs []*IstioCheck) {
 		} else if c.Severity == WarningSeverity {
 			summary.Warnings = summary.Warnings + 1
 		}
+		summary.ObjectCount = summary.ObjectCount + 1
```
Comment (jshaughn) [inline models/istio_validation.go]: I see this is consistent with the above, and it's subjective.  But for me either var++ or  var += 1 is a more clear way to increment.  Certainly still fine as is.

Comment (xeviknal) [reply models/istio_validation.go]: Yep, it is an old habit from ruby times. Ruby core decided not to add this feature. Don't remember the rationale behind. Changed :beer:

**Pull request merged**